lib/Middleware/DeflateMiddleware.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace Aerys\Middleware;
+
+use Aerys\Middleware;
+use Aerys\Request;
+use Aerys\Responder;
+use Aerys\Response;
+use Amp\ByteStream\InMemoryStream;
+use Amp\ByteStream\IteratorStream;
+use Amp\Coroutine;
+use Amp\Producer;
+use Amp\Promise;
+
+class DeflateMiddleware implements Middleware {
+    const MAX_CACHE_SIZE = 1024;
+
+    /**
+     * @var int Minimum body length before body is compressed.
+     * @link http://webmasters.stackexchange.com/questions/31750/what-is-recommended-minimum-object-size-for-deflate-performance-benefits
+     */
+    private $minimumLength = 860;
+
+    /** @var string */
+    private $contentRegex = '#^(?:text/.*+|[^/]*+/xml|[^+]*\+xml|application/(?:json|(?:x-)?javascript))$#i';
+
+    /** @var int Minimum chunk size before being compressed. */
+    private $chunkSize = 8192;
```

Comment:
What's now the source of truth for these options? Options class or here? Currently redundant, but never synchronized.

Reply:
Updating the `Options` class to remove unused settings was on my TODO list. This class obviously needs settings for these options yet.